### PR TITLE
Add readWriteSyncPort to MemPimped

### PR DESCRIPTION
**Pull request by tanhongze**

### CODE REVIEW

File: lib/src/main/scala/spinal/lib/Mem.scala [hunk 1]
```diff
@@ -133,12 +133,25 @@ class MemPimped[T <: Data](mem: Mem[T]) {
     ret.data := ClockDomain.current.withRevertedClockEdge()(mem.readSync(ret.address))
     ret
   }
+
+  def readWriteSyncPort : MemReadWritePort[T] = {
```
Comment (Dolu1990) [inline lib/src/main/scala/spinal/lib/Mem.scala:137]: We may want to add  () after the function name, as in the future we may add optional arguements (ex read during write behaviour, and other stuff)

Comment (Dolu1990) [reply lib/src/main/scala/spinal/lib/Mem.scala:137]: (same for readWriteSyncPortWithMask)

File: lib/src/main/scala/spinal/lib/Mem.scala [hunk 2]
```diff
@@ -174,4 +187,18 @@ case class MemReadPortAsync[T <: Data](dataType : T,addressWidth : Int) extends
...
\ No newline at end of file
+}
+
+case class MemReadWritePort[T <: Data](dataType : T, addressWidth : Int, useMask : Boolean = false) extends Bundle with IMasterSlave{
+  val address = UInt(addressWidth bit)
+  val rdata   = cloneOf(dataType)
+  val wdata   = cloneOf(dataType)
+  val enable  = Bool()
+  val write   = Bool()
+  val mask    = ifGen(useMask)(Bits(widthOf(dataType) bits))
```
[the remark this answers is not in the document]
Comment (Dolu1990) [reply lib/src/main/scala/spinal/lib/Mem.scala:198]: Also, 1 mask bit per word bit isn't properly inferable in most FPGA

**Comment by tanhongze** [comment]
Got it.
By the way, maybe the description for write mask should be put into the Spinal documentation?

**Comment by Dolu1990** [comment]
> By the way, maybe the description for write mask should be put into the Spinal documentation?

Done ^^

**Comment by tanhongze** [comment]
Done.
I am too busy recently, sorry.

**Comment by Dolu1990** [comment]
thanks, no worries :)

**Pull request merged**